### PR TITLE
feat(krm): blueprint - add clusters

### PR DESCRIPTION
**Pull request by janetkuo**

### CODE REVIEW

File: krm/1_clusters/Kptfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+apiVersion: kpt.dev/v1alpha1
+kind: Kptfile
+metadata:
+  name: .
```
Comment (morgante) [inline krm/1_clusters/Kptfile:4]:
```suggestion
  name: cicd_clusters
```

File: krm/1_clusters/Kptfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+packageMetadata:
+  shortDescription: sample description
```
Comment (morgante) [inline krm/1_clusters/Kptfile:6]: Let's add one.

File: krm/1_clusters/Kptfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          values:
+          - marker: ${project-id}
+            ref: '#/definitions/io.k8s.cli.setters.project-id'
+    io.k8s.cli.setters.namespace:
+      description: Namespace of Config Connector resources
+      x-k8s-cli:
+        setter:
+          name: namespace
+          value: YOUR_NAMESPACE
+    io.k8s.cli.setters.project-id:
```
Comment (morgante) [inline krm/1_clusters/Kptfile:23]: Required, right?

File: krm/1_clusters/cluster-dev.yaml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+      disabled: false
+    istioConfig:
+      disabled: true
+    networkPolicyConfig:
+      disabled: false
+  clusterAutoscaling:
+    enabled: false
+  databaseEncryption:
+    state: DECRYPTED
+  defaultMaxPodsPerNode: 110
```
Comment (morgante) [inline krm/1_clusters/cluster-dev.yaml:23]: Do we need to set all these? If we're just repeating defaults we shouldn't include them in the KRM.

File: krm/1_clusters/cluster-dev.yaml [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  enableBinaryAuthorization: true
+  ipAllocationPolicy:
+    clusterSecondaryRangeName: anthos-platform-pods-dev
+    servicesSecondaryRangeName: anthos-platform-services-dev
+  location: us-central1
```
Comment (morgante) [inline krm/1_clusters/cluster-dev.yaml:28]: This should probably be a setter.

File: krm/1_clusters/cluster-dev.yaml [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    dailyMaintenanceWindow:
+      duration: PT4H0M0S
+      startTime: "05:00"
+  monitoringService: monitoring.googleapis.com/kubernetes
+  networkPolicy:
+    enabled: true
+    provider: CALICO
+  networkRef:
+    name: anthos-platform
+  nodeConfig:
```
Comment (morgante) [inline krm/1_clusters/cluster-dev.yaml:40]: Do we need node config here and in the node pool?

File: krm/1_clusters/cluster-dev.yaml (hunk 5 above)
Comment (janetkuo) [reply krm/1_clusters/cluster-dev.yaml]: Should this be a setter per environment? e.g. 4 setters: dev-location, staging-location, prod-location and prod-location

Or per location? e.g. 2 setters: first-region (`us-central1`, which is used by both dev, staging, and prod 1) and second-region (`us-east1`)

Comment (morgante) [reply krm/1_clusters/cluster-dev.yaml]: For now, I think it should be a setter per env.

Hopefully we can come up with a better solution from the DRY config brainstorm going forward. I'm not very happy with the options available now.

Comment (janetkuo) [reply krm/1_clusters/cluster-dev.yaml]: Clusters also references subnetworks whose name implies their location (e.g. `anthos-platform-central`). It'd be tricky to make the location setter here match the subnetwork name, especially when subnetworks are created in the foundation layer. 

We need a better way to solve this too.

Comment (morgante) [reply krm/1_clusters/cluster-dev.yaml]: Agreed, we'll need a better solution for that.

**Pull request merged**